pkg/eventbus/event_bus.go

```diff
@@ -3,23 +3,39 @@ package eventbus
 import (
 	"encoding/json"
 	"fmt"
-
+	"github.com/256dpi/gomqtt/packet"
 	"github.com/kubeedge/kubeedge/beehive/pkg/common/config"
 	"github.com/kubeedge/kubeedge/beehive/pkg/common/log"
 	"github.com/kubeedge/kubeedge/beehive/pkg/core"
 	"github.com/kubeedge/kubeedge/beehive/pkg/core/context"
-
 	"github.com/kubeedge/kubeedge/pkg/eventbus/common/util"
 	mqttBus "github.com/kubeedge/kubeedge/pkg/eventbus/mqtt"
 )
 
+var mqttServer *mqttBus.MqttServer
+
+const (
+	internalMqttMode = iota
+	doubleMqttMode
+	externalMqttMode
+)
+
 // eventbus struct
 type eventbus struct {
-	context *context.Context
+	context  *context.Context
+	mqttMode int
 }
 
 func init() {
-	edgeEventHubModule := eventbus{}
+	mode := config.CONFIG.GetConfigurationByKey("mqtt.mode")
+	if mode == nil {
+		// if the edge.yaml is stale, and dont have mqtt.mode. we continue to use the external mqtt broker remain the same.
+		mode = externalMqttMode
```

Comment:
@kexun , we want default mode to be internal mode or external mode ? i feel internal mode is better as we can remove default dependency on external broker. @rohitsardesai83 @m1093782566 Thoughts on this ?

Reply:
I agree, internal only should be the default

Reply:
```suggestion
		mode = externalMqttMode
```
 as the last release, we use the external mqtt with port 1883 as default. so to be compatible with previous versions, i use the external mqtt and port 1883 as default.  so even if the user dont update his old edge.yaml, it can still run will.